cranelift/codegen/src/isa/aarch64/inst.isle

```diff
@@ -2053,3 +2096,30 @@
 (decl cmeq0 (Reg VectorSize) Reg)
 (rule (cmeq0 rn size)
       (vec_misc (VecMisc2.Cmeq0) rn size))
+
+;; Helper for emitting `MInst.AtomicRMW` instructions.
+(decl lse_atomic_rmw (AtomicRMWOp Value Reg Type) Reg)
+(rule (lse_atomic_rmw op p r_arg2 ty)
+      (let (
+          (r_addr Reg p)
+          (dst WritableReg (temp_writable_reg ty))
+          (_ Unit (emit (MInst.AtomicRMW op r_arg2 dst r_addr ty)))
+        )
+        (writable_reg_to_reg dst)))
```

Comment:
The `writable_reg_to_reg` is unnecessary now with implicit conversions (harmless, just more verbose than needed).

Reply:
@cfallin I depend on this PR in #4034, so both @sparker-arm and I would appreciate any feedback on the ISLE code - it should look good now.